src/main/java/tech/pathtoprogramming/currencyconverter/domain/CurrencyExchangeBoard.java

```diff
@@ -1,16 +1,56 @@
 package tech.pathtoprogramming.currencyconverter.domain;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
 import java.math.BigDecimal;
 
 public class CurrencyExchangeBoard implements ExchangeBoard {
 
 
-    // TODO: call the fixer API when you are ready to integrate with the real API
-    // TODO: Endpoint to get exchange rate - https://api.apilayer.com/fixer/latest?symbols=EUR&base=USD
-    // TODO: Must pass in a header called apikey with the value of your api key when you sign up for free - https://apilayer.com/marketplace/fixer-api
+  @Value("${apiKey}")
+    String    API_KEY ;
+    private final String API_ENDPOINT =
+"https://api.apilayer.com/fixer/convert?to=to&from=from&amount=amount" + API_KEY;
+
+
+    private final RestTemplate restTemplate ;
+    private final ObjectMapper objectMapper ;
+
+    public CurrencyExchangeBoard(RestTemplate restTemplate, ObjectMapper objectMapper) {
+        this.restTemplate = restTemplate;
+        this.objectMapper = objectMapper;
+    }
+
+
+    // TODO: Must pass in a header called apikey
+    //  with the value of your api key when you sign up for free
+    //  - https://apilayer.com/marketplace/fixer-api
     @Override
-    public BigDecimal getExchangeRateFor(Money baseMoney, Currency targetCurrency) {
-        throw new UnsupportedOperationException("need to implement");
+    public BigDecimal getExchangeRateFor(Money baseMoney, Currency targetCurrency) throws IOException {
+
+        String url = API_ENDPOINT + "&base=" + baseMoney + "&symbols=" + targetCurrency;
```

Comment:
I think you're using a different endpoint than I intended. All we want here is the exchange rate given two currencies. That can be provided using the latest endpoint from Fixer.

You can construct the url like below:

```java
        String url = "https://api.apilayer.com/fixer/latest?symbols=" + targetCurrency.name() + "&base=" + baseMoney.currency().name();
        String response = restTemplate.getForObject(url, String.class);
        return objectMapper.readValue(response, BigDecimal.class);
```

The next problem you'll run into is that there is no api key. For Fixer, you'll need to add an HTTP Header with the key being apikey. Unfortunately you can't add headers to the getForObject method from Rest Template. So you'll need to use the method called exchange. Google around and find out how to pass in a header to the RestTemplate exchange method and you should be well on your way to completing all the tasks.